src/app/components/add-volunteers/add-volunteers.component.ts

```diff
@@ -52,80 +57,63 @@ export class AddVolunteersComponent implements OnInit {
-        5: [false],
-        6: [false],
-        7: [false],
-      }),
-      regions: this.fb.array(this.regions.map(region => this.fb.control(false))),
+      multiOccurrence: [''],
+      regions: [null],
     });
   }
 
   sendData() {
     const formValues = this.myForm.value;
     const volunteeringOffer: VolunteeringOffer = this.formValues2volunteeringOfferModel(formValues);
 
-    console.log(volunteeringOffer);
-
     this.vltrOffer.create(volunteeringOffer).then(volunteeringOfferDocument => {
-      console.log(volunteeringOfferDocument);
       if (volunteeringOfferDocument) {
         this.router.navigateByUrl('/dashboard');
       }
     });
   }
 
-  regionCheckboxsToRegionArray(regionCheckboxArray: [boolean]) {
-    const availableRegions: [string] = [] as any;
-    regionCheckboxArray.forEach((regionChecked, i) => {
-      if (regionChecked) {
-        availableRegions.push(this.regions[i]);
-      }
-    });
-
-    return availableRegions;
-  }
-
   formValues2volunteeringOfferModel(formValues): VolunteeringOffer {
+    console.log('forms values: ', formValues);
     const volunteeringOffer: VolunteeringOffer = {} as any;
     volunteeringOffer.title = formValues.title;
     volunteeringOffer.contact = formValues.poc;
     volunteeringOffer.numberOfVolunteers = formValues.numberOfVolunteers;
-    volunteeringOffer.availableContent = formValues.availableContent;
-    volunteeringOffer.availableWeekdays = this.daysCheckboxsToArray(formValues.days);
-    volunteeringOffer.regions = this.regionCheckboxsToRegionArray(formValues.regions);
+    volunteeringOffer.content = formValues.availableContent;
+    if ( formValues.multiOccurrence === '' ) {
+       volunteeringOffer.multiOccurrence = false;
+    } else {
+        volunteeringOffer.multiOccurrence = formValues.multiOccurrence;
+    }
+    volunteeringOffer.regions = formValues.regions;
     volunteeringOffer.createdByUserId = this.auth.getLocalUserId();
-    volunteeringOffer.availableContent = this.content._id;
 
     return volunteeringOffer;
   }
 
-  daysCheckboxsToArray(days): number[] {
-    const availableWeekdays: number[] = [];
-    if (days['1']) { availableWeekdays.push(0); }
-    if (days['2']) { availableWeekdays.push(1); }
-    if (days['3']) { availableWeekdays.push(2); }
-    if (days['4']) { availableWeekdays.push(3); }
-    if (days['5']) { availableWeekdays.push(4); }
-    if (days['6']) { availableWeekdays.push(5); }
-    if (days['7']) { availableWeekdays.push(6); }
-
-    return availableWeekdays;
-  }
-
   openContentModal() {
     this.isFileSelected = false;
     $('#modalContentUpload').modal('toggle');
   }
 
-  onContentTitleLoaded(title: string) {
+  onContentTitleLoaded(changedContent: ContentModel) {
     this.myForm.patchValue({
-      availableContent: title
+      availableContent: changedContent.title
     });
+    this.content = changedContent;
     this.isFileSelected = true;
   }
+
+  groupByFn(regionItem: SubRegionsModel) {
```

Comment:
Maybe rename to `groupByKey`? Your choice.

Reply:
groupByRegion